samples/snippets/src/main/java/pubsub/UpdateDeadLetterPolicyExample.java

```diff
@@ -0,0 +1,96 @@
+ * limitations under the License.
+ */
+
+package pubsub;
+
+// [START pubsub_dead_letter_update_subscription]
+
+import com.google.cloud.pubsub.v1.SubscriptionAdminClient;
+import com.google.protobuf.FieldMask;
+import com.google.pubsub.v1.DeadLetterPolicy;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.Subscription;
+import com.google.pubsub.v1.TopicName;
+import com.google.pubsub.v1.UpdateSubscriptionRequest;
+import java.io.IOException;
+
+public class UpdateDeadLetterPolicyExample {
+  public static void main(String... args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "Your Project ID";
+    // This is an existing subscription with a dead letter policy.
+    String subscriptionId = "Your Subscription ID";
+    // This is an existing topic that the subscription with dead letter policy is attached to.
+    String topicId = "Your Topic ID";
+    // This is an existing dead letter topic that the subscription with dead letter policy forwards
+    // dead letter messages to.
+    String deadLetterTopicId = "Your Dead Letter Topic ID";
+
+    UpdateDeadLetterPolicyExample.updateDeadLetterPolicyExample(
+        projectId, subscriptionId, topicId, deadLetterTopicId);
+  }
+
+  public static void updateDeadLetterPolicyExample(
+      String projectId, String subscriptionId, String topicId, String deadLetterTopicId)
+      throws IOException {
+    try (SubscriptionAdminClient subscriptionAdminClient = SubscriptionAdminClient.create()) {
+
+      ProjectSubscriptionName subscriptionName =
+          ProjectSubscriptionName.of(projectId, subscriptionId);
+
+      System.out.println(
+          "Before: " + subscriptionAdminClient.getSubscription(subscriptionName).getAllFields());
+
+      TopicName topicName = TopicName.of(projectId, topicId);
+      TopicName deadLetterTopicName = TopicName.of(projectId, deadLetterTopicId);
+
+      // Construct the dead letter policy you expect to have after the update.
+      DeadLetterPolicy deadLetterPolicy =
+          DeadLetterPolicy.newBuilder()
+              .setDeadLetterTopic(deadLetterTopicName.toString())
+              .setMaxDeliveryAttempts(20)
+              .build();
+
+      // Construct the subscription with the dead letter policy you expect to have
+      // after the update. Here, values in the required fields (name, topic) help
+      // identify the subscription.
+      Subscription subscription =
+          Subscription.newBuilder()
+              .setName(subscriptionName.toString())
+              .setTopic(topicName.toString())
+              .setDeadLetterPolicy(deadLetterPolicy)
+              .build();
+
+      // Construct a field mask to indicate which field to update in the subscription.
+      FieldMask updateMask =
+          FieldMask.newBuilder().addPaths("dead_letter_policy.max_delivery_attempts").build();
```

Comment:
I think this field mask should also include `dead_letter_policy.dead_letter_topic`

Reply:
I just double-checked. As long as the topic is set here, update will succeed.

https://github.com/googleapis/java-pubsub/blob/44a205b332c54712ad3157d5df57719bbb14948c/samples/snippets/src/main/java/pubsub/UpdateDeadLetterPolicyExample.java#L60-L65